core/src/iosMain/kotlin/com/revenuecat/purchases/kmp/Purchases.ios.kt

```diff
@@ -1,288 +1,370 @@
 
 package com.revenuecat.purchases.kmp
 
 import cocoapods.PurchasesHybridCommon.RCCommonFunctionality
-import cocoapods.PurchasesHybridCommon.RCPurchases
 import cocoapods.PurchasesHybridCommon.RCStoreProduct
+import cocoapods.PurchasesHybridCommon.configureWithAPIKey
+import cocoapods.PurchasesHybridCommon.setAirshipChannelID
+import cocoapods.PurchasesHybridCommon.setOnesignalUserID
 import cocoapods.PurchasesHybridCommon.showStoreMessagesForTypes
+import com.revenuecat.purchases.kmp.models.BillingFeature
 import com.revenuecat.purchases.kmp.models.GoogleReplacementMode
 import com.revenuecat.purchases.kmp.models.PromotionalOffer
 import com.revenuecat.purchases.kmp.models.StoreMessageType
 import com.revenuecat.purchases.kmp.models.StoreProduct
 import com.revenuecat.purchases.kmp.models.StoreProductDiscount
 import com.revenuecat.purchases.kmp.models.StoreTransaction
 import com.revenuecat.purchases.kmp.models.SubscriptionOption
-
-public actual typealias Purchases = RCPurchases
-
-public actual var Purchases.finishTransactions: Boolean
-    get() = finishTransactions()
-    set(value) {
-        setFinishTransactions(value)
+import com.revenuecat.purchases.kmp.strings.ConfigureStrings
+import platform.Foundation.NSURL
+import cocoapods.PurchasesHybridCommon.RCDangerousSettings as IosDangerousSettings
+import cocoapods.PurchasesHybridCommon.RCPurchases as IosPurchases
+
+public actual class Purchases(private val iosPurchases: IosPurchases) {
+    public actual companion object {
+        private var _sharedInstance: Purchases? = null
+        public actual val sharedInstance: Purchases
+            get() = _sharedInstance ?: error(ConfigureStrings.NO_SINGLETON_INSTANCE)
+
+        public actual var logLevel: LogLevel
+            get() = IosPurchases.logLevel().toLogLevel()
+            set(value) = IosPurchases.setLogLevel(value.toRcLogLevel())
+
+        public actual var logHandler: LogHandler
+            get() = IosPurchases.logHandler().toLogHandler()
+            set(value) = IosPurchases.setLogHandler(value.toRcLogHandler())
+
+        public actual var proxyURL: String?
+            get() = IosPurchases.proxyURL()?.absoluteString()
+            set(value) = IosPurchases.setProxyURL(value?.let { NSURL(string = it) })
+
+        public actual val isConfigured: Boolean
+            get() = IosPurchases.isConfigured()
+
+        public actual var simulatesAskToBuyInSandbox: Boolean
+            get() = IosPurchases.simulatesAskToBuyInSandbox()
+            set(value) = IosPurchases.setSimulatesAskToBuyInSandbox(value)
+
+        public actual var forceUniversalAppStore: Boolean
+            get() = IosPurchases.forceUniversalAppStore()
+            set(value) = IosPurchases.setForceUniversalAppStore(value)
+
+        public actual fun configure(
+            configuration: PurchasesConfiguration
+        ): Purchases {
+            // TODO Enable this in SDK-3301 checkCommonVersion()
```

Comment:
Hmm is this comment outdated? Not sure what it refers to (the `checkCommonVersion` method is below)

Reply:
It refers to these PRs 😄 
- https://github.com/RevenueCat/purchases-hybrid-common/pull/838
- https://github.com/RevenueCat/purchases-kmp/pull/104